compiler/executable/insert/executable.rs

```diff
@@ -380,13 +402,18 @@ pub(crate) fn get_kinds_from_types(types: &BTreeSet<Type>) -> HashSet<Kind> {
 }
 
 pub(crate) fn prepare_output_row_schema(
-    input_variables: &HashMap<Variable, VariablePosition>,
+    input_positions: &HashMap<Variable, VariablePosition>,
+    output_positions: &HashMap<Variable, VariablePosition>,
 ) -> Vec<Option<(Variable, VariableSource)>> {
-    let output_width = input_variables.values().map(|i| i.position + 1).max().unwrap_or(0);
+    let output_width = output_positions.values().map(|i| i.position + 1).max().unwrap_or(0);
     let mut output_row_schema = vec![None; output_width as usize];
-    input_variables.iter().map(|(v, i)| (i, v)).for_each(|(&i, &v)| {
-        output_row_schema[i.position as usize] = Some((v, VariableSource::InputVariable(i)));
+    output_positions.iter().for_each(|(var, pos)| {
+        let source =
+            input_positions.get(var).map(|pos| VariableSource::Input(*pos)).unwrap_or(VariableSource::Inserted);
+        output_row_schema[pos.position as usize] = Some((*var, source));
     });
+    // This assumption is made in prepare_output_rows, though it's likely to be violated only for put which doesn't make that assumption.
```

Comment:
Actually, why make that assumption. I'll just make the insert stage also copy-mapped